Implementacion clases Queue y Stack

nahuel1114 opened the pull request.

GabrielEValenzuela committed:
add(pila_cola): Base code

NahuelAroca committed:
Update Tarea

Se implementaron los metodos faltantes en queue y stack. Se crearon los archivos .cpp de list, queue y stack.

NahuelAroca committed:
Update Tarea

Se implementaron metodos print() en queue y stack, tambien metodos registerDecision y onNewEvent en clase Engine. Se agregaron comandos de consola para CHECK_EVENT, SHO_EVENT y SHOW_HISTORY. Se creo un struct Event para registrar los eventos del jugador.

GabrielEValenzuela commented on src/stack.cpp:
```diff
@@ -0,0 +1,83 @@
+        std::cout << temp->data << " ";
+        temp = temp->next;
+    }
+    delete temp;
```
Ojo, hacemos un print, por qué añadis un delete? 👀

nahuel1114 replied:
estoy creando un nodo temporal para poder iterar y hacer temp->data, pero despues de terminar deberia liberar la memoria o no?

nahuel1114 replied:
Ya me di cuenta del error!

GabrielEValenzuela replied:
Genera un cambio mimino y comitealo, veamos que hace el test

GabrielEValenzuela replied:
Porque me parece que asignas esa dirección de memoria para lectura, entonces el delete no es necesario 🤔 
Capaz estoy equivocado

nahuel1114 replied:
Sisi, estoy creando un puntero pero apunta  a direccion de memoria que ya esta creada, con el delete estaria eliminando el ultimo nodo de Stack/Queue. Estaba medio dormido jaja

GabrielEValenzuela commented on src/stack.cpp:
```diff
@@ -0,0 +1,83 @@
+// Created by nahue on 5/4/2025.
+//
+
+#include "../include/dataStructures/stack.hpp"
```
```suggestion
#include "dataStructures/stack.hpp"
```

GabrielEValenzuela commented on src/queue.cpp:
```diff
@@ -0,0 +1,85 @@
+        std::cout << temp->data << " ";
+        temp = temp->next;
+    }
+    delete temp;
```
👀

GabrielEValenzuela reviewed (changes requested):
GJ!

NahuelAroca committed:
Update Tarea

Correccion en metodos print de Queue y Stack.

GabrielEValenzuela commented:
@nahuel1114 falla la compilacion 😢

NahuelAroca committed:
Correccion compilacion.

Me ayudo chatgpt por que iba mas alla de mis conocimientos pero habian problemas de linkeo cuando se creaba un Queue con Event y un List con Refugiado. Tambien daba problemas al tratarlos de imprimir por pantalla, ya que la funcion print() de Queue realizaba un cout << temp->data pero al estar apuntando a un Event no sabia como resolverlo.

nahuel1114 commented:
Buenas, recien pude hacer que compile, ya realice el commit y el push, mi
duda es si se ve en la PR ya creada o debería crear otra? Gracias.

El lun, 7 abr 2025 a las 21:33, GabrielEValenzuela (<
***@***.***>) escribió:

> @nahuel1114 <https://github.com/nahuel1114> falla la compilacion 😢
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GabrielEValenzuela/AlgorithmsAndDataStructures_Course2025/pull/25#issuecomment-2784932431>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BQWF7PWKD6UDZQ5CY5ROBOD2YMKNHAVCNFSM6AAAAAB2U3D6JCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOBUHEZTENBTGE>
> .
> You are receiving this because you were mentioned.Message ID:
> <GabrielEValenzuela/AlgorithmsAndDataStructures_Course2025/pull/25/c2784932431
> @github.com>
> [image: GabrielEValenzuela]*GabrielEValenzuela* left a comment
> (GabrielEValenzuela/AlgorithmsAndDataStructures_Course2025#25)
> <https://github.com/GabrielEValenzuela/AlgorithmsAndDataStructures_Course2025/pull/25#issuecomment-2784932431>
>
> @nahuel1114 <https://github.com/nahuel1114> falla la compilacion 😢
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/GabrielEValenzuela/AlgorithmsAndDataStructures_Course2025/pull/25#issuecomment-2784932431>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BQWF7PWKD6UDZQ5CY5ROBOD2YMKNHAVCNFSM6AAAAAB2U3D6JCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOBUHEZTENBTGE>
> .
> You are receiving this because you were mentioned.Message ID:
> <GabrielEValenzuela/AlgorithmsAndDataStructures_Course2025/pull/25/c2784932431
> @github.com>
>
